### ISSUE TITLE
Add UI element to change font size

### ISSUE THREAD

**Issue by diniamo** [gabm/Satty#106]
The title is self-explanatory, the default font size is way too big for small images.

**Comment by jacekpoz** [comment]
any suggestions on where the UI for that could be and what it could look like?

**Comment by diniamo** [comment]
Probably an input field with +- buttons on the bottom, next to the SML buttons.

**Comment by gabm** [comment]
I agree.. but i don't have a 100% good idea on how to keep the interface lean with the increasing number of options/settings

**Comment by gabm** [comment]
there are some ideas in #42 and #49

**Comment by 00-kat** [comment]
The SML buttons could be replaced with a
`- A +`
when the font tool is selected maybe. The A could be the current font size, instead of showing the number.

I was thinking of maybe sending a PR which replaces SML with - o + where o is the current pen size (after I figure out GTK), so if I ever do that this would fit in even more...

**Comment by diniamo** [comment]
Could work to be honest.

**Comment by RobertMueller2** [comment]
I've had a brief look here. My personal favourite would be to expose annotation-size-factor with 2 digits via a SpinButton in order to make this configurable at run time. It's not a font size, but if you set this to, e.g. 0.25, it's reasonably small. Given how the sizes are currently implemented, I think that is the least invasive approach.

I'm happy to try it, if anyone can help me with a hint where to store this runtime value for later access.

The toolbar side is flexible, it fires an event and can access app_config or style or whatever. But reading it in size impl is the tricky part, because size currently is an enum. that leaves two options IMO:
- overwrite this in configuration (and perhaps add default-annotation-size-factor so it can also be restored to config/cli value easily with a reset button). I tried this but ended up with deadlock from the control, probably can be solved. But since nothing else is doing it like that, I wasn't sure if this really is the way to go
- change Size so it knows whatever object stores the runtime value or convert Size into a struct that holds it. Also didn't really feel right.

Any thoughts on this?
